### ISSUE TITLE
Decaying learn-rate

### ISSUE THREAD

**Issue by dbaumgarten** [dbaumgarten/FToDTF#13]
We are currently using a fixed learn-rate of 1.0 (which is pretty high).
IIRC the paper proposes to continually make the learn-rate smaller while the training progresses.

**Comment by dbaumgarten** [comment]
Postponed until #18 is completed, because to compute the formula for the decaying learn-rate it is necessary to know the current epoch. With the current implementation this is impossible.

**Comment by dbaumgarten** [comment]
Regarding the decaying learnrate the paper states:
> Given a training set containing T words and
> a number of passes over the data equal to P , the
> step size at time t is equal to γ (1 − t/(TP) ), where
> γ is a fixed parameter.

The thing is we don't have a fixed amount of epochs, but a fixed amount of training steps and also use batching instead of computing gradients for every single training sample.

The meaning of the paper seems to be: Start with a learn rate of y, lineariliy reduce the learnrate until it reaches 0 when the training-data has been processed as many times as wanted by the user.

The translated formula for us would be:
y * (1 - t/settings.steps)

This is how we (or I) are going to implement this.
